src/tests/ftest/nvme/enospace.py

```diff
@@ -23,47 +25,65 @@ class NvmeEnospace(ServerFillUp):
     :avocado: recursive
     """
 
+    DER_NOSPACE = -1007
+    DER_TIMEDOUT = -1011
+
     def __init__(self, *args, **kwargs):
         """Initialize a NvmeEnospace object."""
         super().__init__(*args, **kwargs)
+
+        self.expected_errors = [self.DER_NOSPACE, self.DER_TIMEDOUT]
+        self.test_result = []
         self.daos_cmd = None
 
     def setUp(self):
         """Initial setup"""
         super().setUp()
 
+        self.test_result = []
+
         # initialize daos command
         self.daos_cmd = DaosCommand(self.bin)
         self.create_pool_max_size()
-        self.der_nospace_count = 0
-        self.other_errors_count = 0
-        self.test_result = []
 
-    def verify_enospace_log(self, der_nospace_err_count):
-        """
-        Function to verify there are no other error except DER_NOSPACE and
-        DER_NO_HDL in client log.Verify DER_NOSPACE count is higher.
+    def verify_enospace_log(self, log_file):
+        """ Function checking logs consistency.
 
-        args:
-            expected_err_count(int): Expected DER_NOSPACE count from client log.
+        Function checking that only expected errors have occurred and the DER_NOSPACE errors have
+        occurred.
+
+        Args:
+            log_file (string): name prefix of the log files to check.
         """
-        # Get the DER_NOSPACE and other error count from log
-        self.der_nospace_count, self.other_errors_count = error_count(
-            "-1007", self.hostlist_clients, self.client_log)
-
-        # Get the DER_NO_HDL and other error count from log
-        der_nohdl_count, other_nohdl_err = error_count(
-            "-1002", self.hostlist_clients, self.client_log)
-
-        # Check there are no other errors in log file except DER_NO_HDL
-        if self.other_errors_count != der_nohdl_count:
-            self.fail('Found other errors, count {} in client log {}'
-                      .format(int(self.other_errors_count - other_nohdl_err),
-                              self.client_log))
-        # Check the DER_NOSPACE error count is higher if not test will FAIL
-        if self.der_nospace_count < der_nospace_err_count:
-            self.fail('Expected DER_NOSPACE should be > {} and Found {}'
-                      .format(der_nospace_err_count, self.der_nospace_count))
+        logfile_glob = log_file + r".*[0-9]"
```

Comment:
I found the logic of this method hard to understand. I think it would be better if you first check `errors_count[self.DER_NOSPACE]` isn't 0, which you already do in line 74, then compare `errors_count` with `self.expected_errors` as `if errors_count != self.expected_errors:`. If this turns out to be true, then iterate the dictionaries to build the `Found unexpected errors in client logs ...` message.

Reply:
Agree on moving the test `errors_count[self.DER_NOSPACE]` at the beginning of the function.
However, comparing `len(errors_count)` with `len(expected_errors)` could not be used as it for detecting if unexpected errors occurred. Indeed, `errors_count` could contains for example the errors `DER_NOSPACE` and DER_FOO` and the test will pass.  This is why I have to first construct the dict `unexpected_errors_count`.

- [ ] Move errors_count[self.DER_NOSPACE]` test

Reply:
fixed with commit 8fbb62ee8